packages/devtools_app/lib/src/shared/table.dart

```diff
@@ -315,6 +349,23 @@ class FlatTableState<T> extends State<FlatTable<T>>
         secondarySortColumn: secondarySortColumn,
       ),
     );
+    if (widget.pinBehavior != FlatTablePinBehavior.none) {
+      // Collect the list of pinned entries. We don't need to sort again since
+      // we've already sorted the original data.
+      final dataCopy = <T>[];
+      for (final entry in data) {
+        final pinnableEntry = entry as PinnableListEntry;
+        if (pinnableEntry.pinToTop) {
+          pinnedData.add(entry);
+        }
+        if (!pinnableEntry.pinToTop ||
+            widget.pinBehavior == FlatTablePinBehavior.pinCopyToTop) {
+          dataCopy.add(entry);
+        }
+      }
+      data.clear();
+      data.addAll(dataCopy);
```

Comment:
why not just do `data = List.from(dataCopy);`

Reply:
May as well just do `data = dataCopy`. I think this is an artifact from a different approach I was taking before.